DiscordBot/DiscordCodeAlong.md

```diff
@@ -0,0 +1,280 @@
+# Discord Chat Bot
+
+In this activity, create a Discord chat bot that can respond to messages in a Discord server. How the chat bot responds and what it says is for you to decide! The chat bot could respond to common phrases, or respond to commands with useful features like rolling dice, posting pictures, or acting as a calculator.
+
+## Setting Up
+
+1. Go to [repl.it](https://repl.it)
+1. Log in
+1. Clone the [HyTechClub-Robot-Start repl](https://repl.it/@brebory/HyTechClub-Robot-Start)
+1. Make sure you have a Discord account so you can create and register your bot
+
+## Creating and Registering the Bot
+
+The first step is to navigate to the [Discord Developer Portal](https://discord.com/developers/applications). You'll be prompted to log in, so use your Discord account that you created during set up.
+
+1. Make sure that you're on the Applications tab, and then click the blue "New Application" button in the top right
+1. The "Create An Application" dialog will appear, so give your appliction a name, then click Create
+1. You will be taken to your new application's General Information page. In the left-hand sidebar, click on "Bot"
+1. On the Bot page, click the "Add Bot" button
+
+If you get an error about "Too many users share this name", then you will need to change your application's name on the General Information page first, then try again with the new name.
+
+## Saving the Bot Token in the .env File
+
+Repl.it has a special feature that allows you to create environment variables for your Repl application. In your Repl, create a new file called `.env`. In this plaintext file, you can create environment variables by making key-value pairs like this:
+
+```text
+ENVIRONMENT_VARIABLE_NAME=Value
+ANOTHER_VARIABLE=anothervalue
+```
+
+Each environment variable should be on it's own line within this file.
+
+1. In the .env file, create a new variable called `DISCORD_BOT_SECRET`
+1. Back in the Discord Developer Portal, navigate to your bot's page, and find the "Copy" button for your bot's "Token"
+![Discord Bot Settings](discord-bot-settings-cropped.jpg)
+1. Click the copy button and paste the Bot Token into your Repl's `.env` file
+
+### Code
+
+```text
+DISCORD_BOT_SECRET=ABCD.EFGHIJKLMNOPQR.STUVWXYZ
+```
+
+## Adding the Bot to the Sandbox Server
+
+1. If you haven't already created a Discord account, you'll need to for this step
+1. Join the [Hy-Tech Club Chat Bot Sandbox](https://discord.gg/7VUAKkHEvm)
+1. In the `#welcome` channel, click the beaker reaction on MEE6's message to give yourself the Bot Tester role
+1. Copy the "Client ID" from your Discord Application
+![Discord Application Client ID](discord-app-settings-cropped.jpg)
+1. Invite your bot to the Sandbox server by opening your bot's invite link in a browser
+1. Use the following link format, but replace the `client_id` url parameter with your application's client id
+https://discord.com/oauth2/authorize?client_id=xxxxxx&scope=bot

```

Comment:
```suggestion
1. Invite your bot to the Sandbox server by opening your bot's invite link in a browser. Use the following link format, but replace the `client_id` url parameter with your application's client id

    https://discord.com/oauth2/authorize?client_id=xxxxxx&scope=bot

```
Condense to a single element to avoid confusion